OIDC Extension throws ClassNotFoundException when using quarkus-resteasy-jackson

**Describe the bug**
I am migration a project with the Keycloak extension (Quarkus 0.22.0) to the new OIDC extension (Quarkus 0.25.0). I am using the `quarkus-resteasy-jackson` extension instead of jsonb. Unfortunately, these extensions are no longer interchangeable: the access token cannot be parsed, because org.eclipse.microprofile.jwt.Claims specifies JsonObject types, e.g. for the address.
See [stacktrace_JsonObject_notFound.txt](https://github.com/quarkusio/quarkus/files/3744710/stacktrace_JsonObject_notFound.txt)

**Expected behavior**
The app does not crash. Required dependencies are provided.

**Actual behavior**
ClassNotFoundException


**To Reproduce**
Steps to reproduce the behavior:
1. In the `using-openid-connect` starter pom, change the dependency `quarkus-resteasy-jsonb` to `quarkus-resteasy-jackson` 
2. access UserResource as described in the [guide](https://quarkus.io/guides/oidc-guide.html).

For the record, here is the stracktrace:
```
java.lang.NoClassDefFoundError: javax/json/JsonObject
        at org.eclipse.microprofile.jwt.Claims.<clinit>(Claims.java:71)
        at io.smallrye.jwt.auth.principal.DefaultJWTCallerPrincipal.getRawToken(DefaultJWTCallerPrincipal.java:68)
        at io.smallrye.jwt.auth.principal.DefaultJWTCallerPrincipal.<init>(DefaultJWTCallerPrincipal.java:60)
        at io.smallrye.jwt.auth.principal.DefaultJWTCallerPrincipal.<init>(DefaultJWTCallerPrincipal.java:64)
        at io.quarkus.oidc.VertxJwtCallerPrincipal.<init>(VertxJwtCallerPrincipal.java:15)
        at io.quarkus.oidc.VertxOAuth2IdentityProvider$1.handle(VertxOAuth2IdentityProvider.java:62)
        at io.quarkus.oidc.VertxOAuth2IdentityProvider$1.handle(VertxOAuth2IdentityProvider.java:46)
        at io.vertx.ext.auth.oauth2.impl.OAuth2AuthProviderImpl.lambda$decodeToken$3(OAuth2AuthProviderImpl.java:271)
        at io.vertx.ext.auth.oauth2.impl.OAuth2AuthProviderImpl.authenticate(OAuth2AuthProviderImpl.java:238)
        at io.vertx.ext.auth.oauth2.impl.OAuth2AuthProviderImpl.decodeToken(OAuth2AuthProviderImpl.java:269)
        at io.quarkus.oidc.VertxOAuth2IdentityProvider.authenticate(VertxOAuth2IdentityProvider.java:46)
        at io.quarkus.oidc.VertxOAuth2IdentityProvider.authenticate(VertxOAuth2IdentityProvider.java:23)
        at io.quarkus.oidc.VertxOAuth2IdentityProvider_ClientProxy.authenticate(VertxOAuth2IdentityProvider_ClientProxy.zig:152)
        at io.quarkus.security.runtime.QuarkusIdentityProviderManagerImpl.handleProvider(QuarkusIdentityProviderManagerImpl.java:104)
        at io.quarkus.security.runtime.QuarkusIdentityProviderManagerImpl.authenticate(QuarkusIdentityProviderManagerImpl.java:72)
        at io.quarkus.security.runtime.IdentityProviderManagerCreator_ProducerMethod_ipm_91f102be1b2a781216db8a81e6ab4b9b1a84f03c_ClientProxy.authenticate(IdentityProviderManagerCreator_ProducerMethod_ipm_91f102be1b2a781216db8a81e6ab4b9b1a84f03c_ClientProxy.zig:105)
        at io.quarkus.oidc.VertxOAuth2AuthenticationMechanism.authenticate(VertxOAuth2AuthenticationMechanism.java:78)
        at io.quarkus.oidc.VertxOAuth2AuthenticationMechanism_ClientProxy.authenticate(VertxOAuth2AuthenticationMechanism_ClientProxy.zig:41)
        at io.quarkus.vertx.http.runtime.security.HttpAuthenticator.attemptAuthentication(HttpAuthenticator.java:69)
        at io.quarkus.vertx.http.runtime.security.HttpAuthenticator_ClientProxy.attemptAuthentication(HttpAuthenticator_ClientProxy.zig:150)
        at io.quarkus.vertx.http.runtime.security.HttpSecurityRecorder$1.handle(HttpSecurityRecorder.java:34)
        at io.quarkus.vertx.http.runtime.security.HttpSecurityRecorder$1.handle(HttpSecurityRecorder.java:23)
        at io.vertx.ext.web.impl.RouteImpl.handleContext(RouteImpl.java:276)
        at io.vertx.ext.web.impl.RoutingContextImplBase.iterateNext(RoutingContextImplBase.java:124)
        at io.vertx.ext.web.impl.RoutingContextImpl.next(RoutingContextImpl.java:130)
        at io.quarkus.vertx.http.deployment.devmode.VertxHotReplacementSetup$2.handle(VertxHotReplacementSetup.java:79)
        at io.quarkus.vertx.http.deployment.devmode.VertxHotReplacementSetup$2.handle(VertxHotReplacementSetup.java:65)
        at io.vertx.core.impl.ContextImpl.lambda$null$0(ContextImpl.java:330)
        at io.vertx.core.impl.ContextImpl.executeTask(ContextImpl.java:369)
        at io.vertx.core.impl.EventLoopContext.lambda$executeAsync$0(EventLoopContext.java:38)
        at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:163)
        at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:416)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:515)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.ClassNotFoundException: javax.json.JsonObject
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
        at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
        ... 37 more
```

--- a/extensions/oidc/deployment/pom.xml
+++ b/extensions/oidc/deployment/pom.xml
@@ -30,6 +30,10 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-security-deployment</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jsonp-deployment</artifactId>
+        </dependency>
         <!-- Test dependencies -->
         <dependency>
             <groupId>io.quarkus</groupId>

--- a/extensions/oidc/runtime/pom.xml
+++ b/extensions/oidc/runtime/pom.xml
@@ -31,6 +31,10 @@
             <artifactId>quarkus-security</artifactId>
         </dependency>
         <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-jsonp</artifactId>
+        </dependency>
+        <dependency>
             <groupId>io.smallrye</groupId>
             <artifactId>smallrye-jwt</artifactId>
         </dependency>

--- a/integration-tests/oidc/pom.xml
+++ b/integration-tests/oidc/pom.xml
@@ -25,7 +25,7 @@
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>
-            <artifactId>quarkus-resteasy-jsonb</artifactId>
+            <artifactId>quarkus-resteasy-jackson</artifactId>
         </dependency>
         <dependency>
             <groupId>io.quarkus.keycloak</groupId>

--- a/integration-tests/oidc/src/main/java/io/quarkus/it/keycloak/PermissionResource.java
+++ b/integration-tests/oidc/src/main/java/io/quarkus/it/keycloak/PermissionResource.java
@@ -5,6 +5,7 @@ import java.util.Map;
 
 import javax.annotation.security.RolesAllowed;
 import javax.inject.Inject;
+import javax.json.JsonString;
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
@@ -25,8 +26,13 @@ public class PermissionResource {
     public Map<String, Object> permissions() {
         Map<String, Object> claims = new HashMap<>();
         for (String i : jwt.getClaimNames()) {
-            claims.put(i, jwt.claim(i));
+            claims.put(i, fromJsonWebObject(jwt.getClaim(i)));
         }
         return claims;
     }
+
+    private Object fromJsonWebObject(Object claim) {
+        return claim instanceof JsonString ? ((JsonString) claim).getString() : claim != null ? claim.toString() : null;
+    }
+
 }

